TS typing issues

There are some typing issues from our dist files. Such as
```
/// <reference path="../../../global.d.ts" />
```
The global.d.ts is a private file, should not be referenced by dist files.

There are also some typing on private arguments.


If you clone this repo https://github.com/3cp/au2-vite
Run `npm run build`, it will print out all typing issues.

```
huocp@Chunpengs-MacBook-Pro ~/p/au2-vite (main)> npm run build

> au2-vite@0.0.0 build
> tsc && vite build

node_modules/@aurelia/kernel/dist/types/index.d.ts:1:22 - error TS6053: File '/Users/huocp/playground/au2-vite/node_modules/@aurelia/global.d.ts' not found.

1 /// <reference path="../../../global.d.ts" />
                       ~~~~~~~~~~~~~~~~~~~~

node_modules/@aurelia/platform/dist/types/index.d.ts:1:22 - error TS6053: File '/Users/huocp/playground/au2-vite/node_modules/@aurelia/global.d.ts' not found.

1 /// <reference path="../../../global.d.ts" />
                       ~~~~~~~~~~~~~~~~~~~~

node_modules/@aurelia/router-lite/dist/types/instructions.d.ts:94:102 - error TS2552: Cannot find name 'ViewportInstruction'. Did you mean 'IViewportInstruction'?

94 export interface ITypedNavigationInstruction_ViewportInstruction extends ITypedNavigationInstruction<ViewportInstruction, NavigationInstructionType.ViewportInstruction> {
                                                                                                        ~~~~~~~~~~~~~~~~~~~

node_modules/@aurelia/router-lite/dist/types/route-context.d.ts:5:41 - error TS2724: '"./instructions"' has no exported member named 'ViewportInstruction'. Did you mean 'IViewportInstruction'?

5 import { NavigationInstruction, Params, ViewportInstruction } from './instructions';
                                          ~~~~~~~~~~~~~~~~~~~

node_modules/@aurelia/router-lite/dist/types/route-context.d.ts:19:73 - error TS2304: Cannot find name 'EagerInstruction'.

19 export declare function isEagerInstruction(val: NavigationInstruction | EagerInstruction): val is EagerInstruction;
                                                                           ~~~~~~~~~~~~~~~~

node_modules/@aurelia/router-lite/dist/types/route-context.d.ts:19:99 - error TS2304: Cannot find name 'EagerInstruction'.

19 export declare function isEagerInstruction(val: NavigationInstruction | EagerInstruction): val is EagerInstruction;
                                                                                                     ~~~~~~~~~~~~~~~~

node_modules/@aurelia/router-lite/dist/types/route-context.d.ts:103:70 - error TS2304: Cannot find name 'EagerInstruction'.

103     generateViewportInstruction(instruction: NavigationInstruction | EagerInstruction): PathGenerationResult | null;
                                                                         ~~~~~~~~~~~~~~~~

node_modules/@aurelia/router-lite/dist/types/route-expression.d.ts:1:10 - error TS2305: Module '"./instructions"' has no exported member 'ViewportInstructionTree'.

1 import { ViewportInstructionTree, ViewportInstruction, Params } from './instructions';
           ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@aurelia/router-lite/dist/types/route-expression.d.ts:1:35 - error TS2724: '"./instructions"' has no exported member named 'ViewportInstruction'. Did you mean 'IViewportInstruction'?

1 import { ViewportInstructionTree, ViewportInstruction, Params } from './instructions';
                                    ~~~~~~~~~~~~~~~~~~~

node_modules/@aurelia/router-lite/dist/types/route-tree.d.ts:3:65 - error TS2724: '"./instructions"' has no exported member named 'ViewportInstruction'. Did you mean 'IViewportInstruction'?

3 import { ITypedNavigationInstruction_ResolvedComponent, Params, ViewportInstruction, ViewportInstructionTree } from './instructions';
                                                                  ~~~~~~~~~~~~~~~~~~~

node_modules/@aurelia/router-lite/dist/types/route-tree.d.ts:3:86 - error TS2305: Module '"./instructions"' has no exported member 'ViewportInstructionTree'.

3 import { ITypedNavigationInstruction_ResolvedComponent, Params, ViewportInstruction, ViewportInstructionTree } from './instructions';
                                                                                       ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@aurelia/router-lite/dist/types/router-events.d.ts:2:10 - error TS2305: Module '"./instructions"' has no exported member 'ViewportInstructionTree'.

2 import { ViewportInstructionTree } from './instructions';
           ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@aurelia/router-lite/dist/types/router.d.ts:9:73 - error TS2305: Module '"./instructions"' has no exported member 'ViewportInstructionTree'.

9 import { IViewportInstruction, NavigationInstruction, RouteContextLike, ViewportInstructionTree, Params } from './instructions';
                                                                          ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@aurelia/runtime-html/dist/types/binding/listener.d.ts:24:179 - error TS2304: Cannot find name 'ListenerOptions'.

24     constructor(platform: IPlatform, targetEvent: string, sourceExpression: IsBindingBehavior, target: Node, eventDelegator: IEventDelegator, locator: IServiceLocator, _options: ListenerOptions);
                                                                                                                                                                                     ~~~~~~~~~~~~~~~

node_modules/@aurelia/runtime-html/dist/types/index.d.ts:1:22 - error TS6053: File '/Users/huocp/playground/au2-vite/node_modules/@aurelia/global.d.ts' not found.

1 /// <reference path="../../../global.d.ts" />
                       ~~~~~~~~~~~~~~~~~~~~

node_modules/@aurelia/runtime/dist/types/index.d.ts:1:22 - error TS6053: File '/Users/huocp/playground/au2-vite/node_modules/@aurelia/global.d.ts' not found.

1 /// <reference path="../../../global.d.ts" />
                       ~~~~~~~~~~~~~~~~~~~~


Found 16 errors in 11 files.

Errors  Files
     1  node_modules/@aurelia/kernel/dist/types/index.d.ts:1
     1  node_modules/@aurelia/platform/dist/types/index.d.ts:1
     1  node_modules/@aurelia/router-lite/dist/types/instructions.d.ts:94
     4  node_modules/@aurelia/router-lite/dist/types/route-context.d.ts:5
     2  node_modules/@aurelia/router-lite/dist/types/route-expression.d.ts:1
     2  node_modules/@aurelia/router-lite/dist/types/route-tree.d.ts:3
     1  node_modules/@aurelia/router-lite/dist/types/router-events.d.ts:2
     1  node_modules/@aurelia/router-lite/dist/types/router.d.ts:9
     1  node_modules/@aurelia/runtime-html/dist/types/binding/listener.d.ts:24
     1  node_modules/@aurelia/runtime-html/dist/types/index.d.ts:1
     1  node_modules/@aurelia/runtime/dist/types/index.d.ts:1
```

Latest dev build still has few typing issues.

```
huocp@Chunpengs-MacBook-Pro ~/p/au2-vite (main)> npm run build

> au2-vite@0.0.0 build
> tsc && vite build

node_modules/@aurelia/router-lite/dist/types/instructions.d.ts:94:102 - error TS2552: Cannot find name 'ViewportInstruction'. Did you mean 'IViewportInstruction'?

94 export interface ITypedNavigationInstruction_ViewportInstruction extends ITypedNavigationInstruction<ViewportInstruction, NavigationInstructionType.ViewportInstruction> {
                                                                                                        ~~~~~~~~~~~~~~~~~~~

node_modules/@aurelia/router-lite/dist/types/route-context.d.ts:5:41 - error TS2724: '"./instructions"' has no exported member named 'ViewportInstruction'. Did you mean 'IViewportInstruction'?

5 import { NavigationInstruction, Params, ViewportInstruction } from './instructions';
                                          ~~~~~~~~~~~~~~~~~~~

node_modules/@aurelia/router-lite/dist/types/route-context.d.ts:19:73 - error TS2304: Cannot find name 'EagerInstruction'.

19 export declare function isEagerInstruction(val: NavigationInstruction | EagerInstruction): val is EagerInstruction;
                                                                           ~~~~~~~~~~~~~~~~

node_modules/@aurelia/router-lite/dist/types/route-context.d.ts:19:99 - error TS2304: Cannot find name 'EagerInstruction'.

19 export declare function isEagerInstruction(val: NavigationInstruction | EagerInstruction): val is EagerInstruction;
                                                                                                     ~~~~~~~~~~~~~~~~

node_modules/@aurelia/router-lite/dist/types/route-context.d.ts:103:70 - error TS2304: Cannot find name 'EagerInstruction'.

103     generateViewportInstruction(instruction: NavigationInstruction | EagerInstruction): PathGenerationResult | null;
                                                                         ~~~~~~~~~~~~~~~~

node_modules/@aurelia/router-lite/dist/types/route-expression.d.ts:1:10 - error TS2305: Module '"./instructions"' has no exported member 'ViewportInstructionTree'.

1 import { ViewportInstructionTree, ViewportInstruction, Params } from './instructions';
           ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@aurelia/router-lite/dist/types/route-expression.d.ts:1:35 - error TS2724: '"./instructions"' has no exported member named 'ViewportInstruction'. Did you mean 'IViewportInstruction'?

1 import { ViewportInstructionTree, ViewportInstruction, Params } from './instructions';
                                    ~~~~~~~~~~~~~~~~~~~

node_modules/@aurelia/router-lite/dist/types/route-tree.d.ts:3:65 - error TS2724: '"./instructions"' has no exported member named 'ViewportInstruction'. Did you mean 'IViewportInstruction'?

3 import { ITypedNavigationInstruction_ResolvedComponent, Params, ViewportInstruction, ViewportInstructionTree } from './instructions';
                                                                  ~~~~~~~~~~~~~~~~~~~

node_modules/@aurelia/router-lite/dist/types/route-tree.d.ts:3:86 - error TS2305: Module '"./instructions"' has no exported member 'ViewportInstructionTree'.

3 import { ITypedNavigationInstruction_ResolvedComponent, Params, ViewportInstruction, ViewportInstructionTree } from './instructions';
                                                                                       ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@aurelia/router-lite/dist/types/router-events.d.ts:2:10 - error TS2305: Module '"./instructions"' has no exported member 'ViewportInstructionTree'.

2 import { ViewportInstructionTree } from './instructions';
           ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@aurelia/router-lite/dist/types/router.d.ts:9:73 - error TS2305: Module '"./instructions"' has no exported member 'ViewportInstructionTree'.

9 import { IViewportInstruction, NavigationInstruction, RouteContextLike, ViewportInstructionTree, Params } from './instructions';
                                                                          ~~~~~~~~~~~~~~~~~~~~~~~


Found 11 errors in 6 files.

Errors  Files
     1  node_modules/@aurelia/router-lite/dist/types/instructions.d.ts:94
     4  node_modules/@aurelia/router-lite/dist/types/route-context.d.ts:5
     2  node_modules/@aurelia/router-lite/dist/types/route-expression.d.ts:1
     2  node_modules/@aurelia/router-lite/dist/types/route-tree.d.ts:3
     1  node_modules/@aurelia/router-lite/dist/types/router-events.d.ts:2
     1  node_modules/@aurelia/router-lite/dist/types/router.d.ts:9
```

--- a/packages/router-lite/src/instructions.ts
+++ b/packages/router-lite/src/instructions.ts
@@ -72,7 +72,6 @@ export interface IViewportInstruction {
   readonly recognizedRoute: $RecognizedRoute | null;
 }
 
-/** @internal */
 export class ViewportInstruction<TComponent extends ITypedNavigationInstruction_T = ITypedNavigationInstruction_Component> implements IViewportInstruction {
   private constructor(
     public open: number,
@@ -267,7 +266,6 @@ const getObjectId = (function () {
   };
 })();
 
-/** @internal */
 export class ViewportInstructionTree {
   public constructor(
     public readonly options: NavigationOptions,

--- a/packages/router-lite/src/route-context.ts
+++ b/packages/router-lite/src/route-context.ts
@@ -21,7 +21,6 @@ export const RESIDUE = 'au$residue' as const;
 
 type PathGenerationResult = { vi: ViewportInstruction; query: Params | null };
 
-/** @internal */
 export type EagerInstruction = {
   component: string | RouteDefinition | PartialCustomElementDefinition | IRouteViewModel | IChildRouteConfig | RouteType;
   params: Params;

